specs/electra/fork.md

```diff
@@ -0,0 +1,144 @@
+        gas_limit=pre.latest_execution_payload_header.gas_limit,
+        gas_used=pre.latest_execution_payload_header.gas_used,
+        timestamp=pre.latest_execution_payload_header.timestamp,
+        extra_data=pre.latest_execution_payload_header.extra_data,
+        base_fee_per_gas=pre.latest_execution_payload_header.base_fee_per_gas,
+        block_hash=pre.latest_execution_payload_header.block_hash,
+        transactions_root=pre.latest_execution_payload_header.transactions_root,
+        withdrawals_root=pre.latest_execution_payload_header.withdrawals_root,
+        blob_gas_used=pre.latest_execution_payload_header.blob_gas_used,  # [Modified in Electra]
+        excess_blob_gas=pre.latest_execution_payload_header.excess_blob_gas,  # [Modified in Electra]
+    )
+    post = BeaconState(
+        # Versioning
+        genesis_time=pre.genesis_time,
+        genesis_validators_root=pre.genesis_validators_root,
+        slot=pre.slot,
+        fork=Fork(
+            previous_version=pre.fork.current_version,
+            current_version=ELECTRA_FORK_VERSION,  # [Modified in Electra]
+            epoch=epoch,
+        ),
+        # History
+        latest_block_header=pre.latest_block_header,
+        block_roots=pre.block_roots,
+        state_roots=pre.state_roots,
+        historical_roots=pre.historical_roots,
+        # Eth1
+        eth1_data=pre.eth1_data,
+        eth1_data_votes=pre.eth1_data_votes,
+        eth1_deposit_index=pre.eth1_deposit_index,
+        # Registry
+        validators=pre.validators,
+        balances=pre.balances,
+        # Randomness
+        randao_mixes=pre.randao_mixes,
+        # Slashings
+        slashings=pre.slashings,
+        # Participation
+        previous_epoch_participation=pre.previous_epoch_participation,
+        current_epoch_participation=pre.current_epoch_participation,
+        # Finality
+        justification_bits=pre.justification_bits,
+        previous_justified_checkpoint=pre.previous_justified_checkpoint,
+        current_justified_checkpoint=pre.current_justified_checkpoint,
+        finalized_checkpoint=pre.finalized_checkpoint,
+        # Inactivity
+        inactivity_scores=pre.inactivity_scores,
+        # Sync
+        current_sync_committee=pre.current_sync_committee,
+        next_sync_committee=pre.next_sync_committee,
+        # Execution-layer
+        latest_execution_payload_header=latest_execution_payload_header,
+        # Withdrawals
+        next_withdrawal_index=pre.next_withdrawal_index,
+        next_withdrawal_validator_index=pre.next_withdrawal_validator_index,
+        # Deep history valid from Capella onwards
+        historical_summaries=pre.historical_summaries,
+        # Sync history
+        previous_best_sync_data=default_sync_data(),  # [New in Electra]
+        current_best_sync_data=default_sync_data(),  # [New in Electra]
+        parent_block_has_sync_committee_finality=False,  # [New in Electra]
```

Comment:
Not ideal, but requires knowledge of the `finalized_checkpoint` in the latest block's immediate post-state, which may be different in case of realization of new finality during empty slots.

For Genesis from Electra onward, could also initialize to `True`.

Not sure what's best, if we initialize to `False` here it just means that `compute_start_slot_at_epoch(ELECTRA_FORK_EPOCH) + 1` would never be treated as the best sync data. Slightly incorrect, but if everyone does it like that, it's consistent and that's what this PR is about.